### PR TITLE
pixinsight: 1.8.9-2-20230920 -> 1.8.9-3-20240619

### PR DESCRIPTION
**Pull request by sheepforce**
## Description of changes

https://pixinsight.com/forum/index.php?threads/pixinsight-1-8-9-2-build-1588-released.21898/

The previous version has been removed from the software distribution and can no longer be downloaded :/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by atalii — APPROVED** [review]
LGTM; Should this be backported to 23.05?

**Comment by sheepforce** [comment]
> LGTM; Should this be backported to 23.05?

Ah this would actually be a good idea, yes! :)

**Review by JohnRTitor — COMMENTED** [review]
Hi, could you update to the [latest version](https://pixinsight.com/forum/index.php?threads/pixinsight-1-8-9-3-released.23507/)?

**Comment by sheepforce** [comment]
> Hi, could you update to the [latest version](https://pixinsight.com/forum/index.php?threads/pixinsight-1-8-9-3-released.23507/)?

Hey :) That is unfortunately not that easy. The last two versions started checking if they are installed to `/opt` and allow for no other option anymore. I've tried to create a bubblewrap around the installer to make it think it is installing to `/opt` instead of `$out` but wasn't successful so far. Possibly it needs to be ported to an FHS env. I will try again, though.

**Comment by sheepforce** [comment]
I've got it working :sweat_smile: Thanks for the heads up @JohnRTitor It involves two more or less hacky solutions but they are required to make it work in a Nix store. Ready to be merged from my perspective.

Changelogs are here: https://pixinsight.com/forum/index.php?threads/pixinsight-1-8-9-3-released.23507/

File: pkgs/applications/graphics/pixinsight/default.nix [hunk 1]
```diff
@@ -128,7 +185,7 @@ stdenv.mkDerivation rec {
     license = licenses.unfree;
     platforms = [ "x86_64-linux" ];
     maintainers = [ maintainers.sheepforce ];
-    hydraPlatforms = [];
+    hydraPlatforms = [ ];
```
Comment (JohnRTitor) [inline pkgs/applications/graphics/pixinsight/default.nix:188]: Why is this here?

File: pkgs/applications/graphics/pixinsight/default.nix [hunk 2]
```diff
@@ -1,18 +1,60 @@
...
+, qtwebview
+, qtscxml
 }:
 
 stdenv.mkDerivation rec {
   pname = "pixinsight";
-  version = "1.8.9-2";
+  version = "1.8.9-3";
 
   src = requireFile rec {
```
Comment (JohnRTitor) [inline pkgs/applications/graphics/pixinsight/default.nix:54]: Instead of using `requireFile`, can we fetch the binary directly?

File: pkgs/applications/graphics/pixinsight/default.nix [hunk 3]
```diff
@@ -1,18 +1,60 @@
...
+, mlt
+, qtlocation
+, qtwebengine
+, qtquick3d
+, qtwayland
+, qtwebview
+, qtscxml
 }:
 
 stdenv.mkDerivation rec {
```
Comment (JohnRTitor) [inline pkgs/applications/graphics/pixinsight/default.nix:50]: `rec`ursion is not preferred, could you move it to `finalAttrs`?

File: pkgs/applications/graphics/pixinsight/default.nix (hunk 2 above)
Comment (sheepforce) [reply pkgs/applications/graphics/pixinsight/default.nix:54]: No, it requires a commercial License and login to the software distribution forum.

File: pkgs/applications/graphics/pixinsight/default.nix (hunk 1 above)
Comment (sheepforce) [reply pkgs/applications/graphics/pixinsight/default.nix:188]: Has been added some time ago by someone to avoid Hydra attempts of building this package and subsequently fail. This is what `lib.meta.dontDistribute` does and is correct for this package.
https://github.com/NixOS/nixpkgs/blob/938aa157bbd6e3c6fd7dcb77998b1f92c2ad1631/lib/meta.nix#L26

Comment (JohnRTitor) [reply pkgs/applications/graphics/pixinsight/default.nix:188]: Hydra won't even try to build `unfree` packages, FYI.

**Review by JohnRTitor — APPROVED** [review]
Let's merge this one after the CI passes

File: pkgs/applications/graphics/pixinsight/default.nix (hunk 1 above)
Comment (sheepforce) [reply pkgs/applications/graphics/pixinsight/default.nix:188]: See here for the reason: https://github.com/NixOS/nixpkgs/pull/190511
